src/Microsoft.NetCore.Analyzers/Core/Security/DoNotCallDangerousMethodsInDeserialization.cs

```diff
@@ -195,24 +233,36 @@ public sealed override void Initialize(AnalysisContext context)
                     // <param name="methodSymbol">The symbol of the method to be analyzed</param>
                     // <param name="visited">All the method has been analyzed</param>
                     // <param name="results">The result is organized by &lt;method to be analyzed, dangerous method it calls&gt;</param>
-                    void FindCalledDangerousMethod(IMethodSymbol methodSymbol, HashSet<IMethodSymbol> visited, Dictionary<IMethodSymbol, HashSet<IMethodSymbol>> results)
+                    void FindCalledDangerousMethod(ISymbol methodSymbol, HashSet<ISymbol> visited, Dictionary<ISymbol, HashSet<ISymbol>> results)
                     {
                         if (visited.Add(methodSymbol))
                         {
-                            results.Add(methodSymbol, new HashSet<IMethodSymbol>());
+                            results.Add(methodSymbol, new HashSet<ISymbol>());
+
+                            if (!callGraph.TryGetValue(methodSymbol, out var calledMethods))
+                            {
+                                Debug.Fail(methodSymbol.Name + " was not found in callGraph");
+
+                                return;
+                            }
 
-                            foreach (var child in callGraph[methodSymbol].Keys)
+                            foreach (var child in calledMethods.Keys)
                             {
                                 if (dangerousMethodSymbols.Contains(child))
                                 {
                                     results[methodSymbol].Add(child);
                                 }
 
-                                if (child.IsInSource())
+                                FindCalledDangerousMethod(child, visited, results);
+
+                                if (!results.TryGetValue(child, out var result))
                                 {
-                                    FindCalledDangerousMethod(child, visited, results);
-                                    results[methodSymbol].UnionWith(results[child]);
+                                    Debug.Fail(child.Name + " was not found in callGraph");
+
+                                    return;
                                 }
+
+                                results[methodSymbol].UnionWith(result);
```

Comment:
I think here, instead of returning, you could just ignore that particular child if it's not found:

```csharp
if (results.TryGetValue(child, out var result))
{
    results[methodSymbol].UnionWith(result);
}
else
{
    Debug.Fail(child.Name + " was not found in callGraph");
}
```